File: frontend/src/types/Course.ts
import type { Difficulty } from './Question'

export interface CoursePublic {
  id: number
  name: string
  description: string | null
}

export interface WeekSummary {
  id: number
  week_number: number
  title: string
  question_count: number
  difficulties: Difficulty[]
}

File: backend/app/schemas/course.py
from pydantic import BaseModel, ConfigDict

from app.models.question import Difficulty


class CoursePublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: str | None


class WeekSummary(BaseModel):
    id: int
    week_number: int
    title: str
    question_count: int
    difficulties: list[Difficulty]
